### PR TITLE
Implement radio input for user id in the claims  field

### PR DESCRIPTION
**Pull request by dasuni-30**
https://github.com/wso2/identity-apps/assets/27746285/1cb2c32a-60a6-4232-920b-c4da842dc004

### Related Issues
- https://github.com/wso2/product-is/issues/19371

### Related PRs
- None

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/features/connections/components/edit/forms/helpers/form-fields-helper.tsx [hunk 1]
```diff
@@ -178,6 +178,48 @@ export const getRadioButtonField = (eachProp: CommonPluggableComponentPropertyIn
...
+                value={ eachProp?.value }
+                requiredErrorMessage={ I18n.instance.t("console:develop.features.authenticationProvider.forms.common." +
+                    "requiredErrorMessage") }
+                children={ options }
+                listen={ (values: Map<string, FormValue>) => {
+                    listen(propertyMetadata?.key, values);
+                } }
+                disabled={ propertyMetadata?.isDisabled }
+                readOnly={ propertyMetadata?.readOnly }
+                data-testid={ `${ testId }-${ propertyMetadata?.key }` }
```
Comment (Achintha444) [inline apps/console/src/features/connections/components/edit/forms/helpers/form-fields-helper.tsx:214]: add `data-componentid` as well

Comment (Achintha444) [reply apps/console/src/features/connections/components/edit/forms/helpers/form-fields-helper.tsx:214]: check other places

File: apps/console/src/features/connections/components/edit/forms/components/common-pluggable-component-form.tsx [hunk 2]
```diff
@@ -77,7 +77,11 @@ export const CommonPluggableComponentForm: FunctionComponent<CommonPluggableComp
     const interpretValueByType = (value: FormValue, key: string, type: string) => {
         switch (type?.toUpperCase()) {
             case CommonConstants.BOOLEAN: {
-                return value?.includes(key);
+                if ( key === ConnectionManagementConstants.USER_ID_IN_CLAIMS) {
```
Comment (JayaShakthi97) [inline apps/console/src/features/connections/components/edit/forms/components/common-pluggable-component-form.tsx:80]:
```suggestion
                if (key === ConnectionManagementConstants.USER_ID_IN_CLAIMS) {
```

File: apps/console/src/features/connections/components/edit/forms/helpers/form-fields-helper.tsx [hunk 3]
```diff
@@ -178,6 +178,48 @@ export const getRadioButtonField = (eachProp: CommonPluggableComponentPropertyIn
     );
 };
 
+// Add a radio field for user id in claims with predefined options.
+export const getUserIdClaimRadioButtonField = (eachProp: CommonPluggableComponentPropertyInterface,
+    propertyMetadata: CommonPluggableComponentMetaPropertyInterface,
+    listen: (key: string, values: Map<string, FormValue>) => void,
+    testId?: string): ReactElement => {
```
Comment (JayaShakthi97) [inline apps/console/src/features/connections/components/edit/forms/helpers/form-fields-helper.tsx:185]: We can improve formatting here.
```suggestion
export const getUserIdClaimRadioButtonField = (
    eachProp: CommonPluggableComponentPropertyInterface,
    propertyMetadata: CommonPluggableComponentMetaPropertyInterface,
    listen: (key: string, values: Map<string, FormValue>) => void,
    testId?: string
): ReactElement => {
```

File: apps/console/src/features/connections/components/edit/forms/helpers/form-fields-helper.tsx [hunk 4]
```diff
@@ -617,7 +659,9 @@ export const getFieldType = (
     propertyMetadata: CommonPluggableComponentMetaPropertyInterface,
     mode: AuthenticatorSettingsFormModes
 ): FieldType => {
-    if (propertyMetadata?.type?.toUpperCase() === CommonConstants.BOOLEAN) {
+    if (propertyMetadata?.key === ConnectionManagementConstants.USER_ID_IN_CLAIMS) {
+        return FieldType.RADIO;
+    } if (propertyMetadata?.type?.toUpperCase() === CommonConstants.BOOLEAN) {
```
Comment (JayaShakthi97) [inline apps/console/src/features/connections/components/edit/forms/helpers/form-fields-helper.tsx:665]: Should this be a `else if`?
If not add a newline here

File: apps/console/src/features/connections/components/edit/forms/helpers/form-fields-helper.tsx (hunk 1 above)
Comment (dasuni-30) [reply apps/console/src/features/connections/components/edit/forms/helpers/form-fields-helper.tsx:214]: Added with d5999789afb51def962c8d959ea0227a28787a30.

File: apps/console/src/features/identity-providers/components/forms/components/common-pluggable-component-form.tsx [hunk 5]
```diff
@@ -105,7 +106,11 @@ export const CommonPluggableComponentForm: FunctionComponent<CommonPluggableComp
     const interpretValueByType = (value: FormValue, key: string, type: string) => {
         switch (type?.toUpperCase()) {
             case CommonConstants.BOOLEAN: {
-                return value?.includes(key);
+                if ( key === IdentityProviderConstants.USER_ID_IN_CLAIMS) {
```
Comment (JayaShakthi97) [inline apps/console/src/features/identity-providers/components/forms/components/common-pluggable-component-form.tsx:109]:
```suggestion
                if (key === IdentityProviderConstants.USER_ID_IN_CLAIMS) {
```

File: apps/console/src/features/identity-providers/components/forms/helpers/form-fields-helper.tsx [hunk 6]
```diff
@@ -186,6 +187,48 @@ export const getRadioButtonField = (eachProp: CommonPluggableComponentPropertyIn
     );
 };
 
+// Add a radio field for user id in claims with predefined options.
+export const getUserIdClaimRadioButtonField = (eachProp: CommonPluggableComponentPropertyInterface,
+    propertyMetadata: CommonPluggableComponentMetaPropertyInterface,
+    listen: (key: string, values: Map<string, FormValue>) => void,
+    testId?: string): ReactElement => {
+    const options: StrictRadioChild[] = [
```
Comment (JayaShakthi97) [inline apps/console/src/features/identity-providers/components/forms/helpers/form-fields-helper.tsx:195]:
```suggestion
export const getUserIdClaimRadioButtonField = (
    eachProp: CommonPluggableComponentPropertyInterface,
    propertyMetadata: CommonPluggableComponentMetaPropertyInterface,
    listen: (key: string, values: Map<string, FormValue>) => void,
    testId?: string
): ReactElement => {
    const options: StrictRadioChild[] = [
```

File: apps/console/src/features/connections/components/edit/forms/components/common-pluggable-component-form.tsx (hunk 2 above)
Comment (dasuni-30) [reply apps/console/src/features/connections/components/edit/forms/components/common-pluggable-component-form.tsx]: Improved with 5047722b4b25f0551dd0da7ca417f606a3ce039a.

File: apps/console/src/features/connections/components/edit/forms/helpers/form-fields-helper.tsx (hunk 3 above)
Comment (dasuni-30) [reply apps/console/src/features/connections/components/edit/forms/helpers/form-fields-helper.tsx]: Improved with 5047722b4b25f0551dd0da7ca417f606a3ce039a.

File: apps/console/src/features/identity-providers/components/forms/components/common-pluggable-component-form.tsx (hunk 5 above)
Comment (dasuni-30) [reply apps/console/src/features/identity-providers/components/forms/components/common-pluggable-component-form.tsx]: Improved with 5047722b4b25f0551dd0da7ca417f606a3ce039a.

File: apps/console/src/features/identity-providers/components/forms/helpers/form-fields-helper.tsx (hunk 6 above)
Comment (dasuni-30) [reply apps/console/src/features/identity-providers/components/forms/helpers/form-fields-helper.tsx]: Improved with 5047722b4b25f0551dd0da7ca417f606a3ce039a.

File: apps/console/src/features/connections/components/edit/forms/helpers/form-fields-helper.tsx (hunk 4 above)
Comment (dasuni-30) [reply apps/console/src/features/connections/components/edit/forms/helpers/form-fields-helper.tsx]: Improved with 5047722b4b25f0551dd0da7ca417f606a3ce039a.

File: apps/console/src/features/connections/components/edit/forms/helpers/form-fields-helper.tsx [hunk 7]
```diff
@@ -178,6 +178,50 @@ export const getRadioButtonField = (eachProp: CommonPluggableComponentPropertyIn
     );
 };
 
+// Add a radio field for user id in claims with predefined options.
```
Comment (brionmario) [inline apps/console/src/features/connections/components/edit/forms/helpers/form-fields-helper.tsx:181]: Better to have a TSDoc comment.

File: apps/console/src/features/identity-providers/constants/identity-provider-constants.ts [hunk 8]
```diff
@@ -1,5 +1,5 @@
 /**
- * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com). All Rights Reserved.
+ * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com).
```
Comment (brionmario) [inline apps/console/src/features/identity-providers/constants/identity-provider-constants.ts:2]:
```suggestion
 * Copyright (c) 2023-2024, WSO2 LLC. (https://www.wso2.com).
```

File: apps/console/src/features/identity-providers/constants/identity-provider-constants.ts [hunk 9]
```diff
@@ -110,6 +110,11 @@ export class IdentityProviderConstants {
      */
     public static readonly JWKS_URL_MIN_LENGTH: number = 10;
 
+    /**
+     * Key for user id in claims.
+     */
+    public static readonly USER_ID_IN_CLAIMS: string = "IsUserIdInClaims";
+
```
Comment (brionmario) [inline apps/console/src/features/identity-providers/constants/identity-provider-constants.ts:117]:
```suggestion\
```

**Review by brionmario — APPROVED** [review]

File: apps/console/src/features/connections/components/edit/forms/helpers/form-fields-helper.tsx (hunk 7 above)
Comment (dasuni-30) [reply apps/console/src/features/connections/components/edit/forms/helpers/form-fields-helper.tsx]: Improved with 520dba0695d501fe0ca61d7edb365d58ea7cf899.

File: apps/console/src/features/identity-providers/constants/identity-provider-constants.ts (hunk 9 above)
Comment (dasuni-30) [reply apps/console/src/features/identity-providers/constants/identity-provider-constants.ts]: Improved with 520dba0695d501fe0ca61d7edb365d58ea7cf899.

File: apps/console/src/features/identity-providers/constants/identity-provider-constants.ts (hunk 8 above)
Comment (dasuni-30) [reply apps/console/src/features/identity-providers/constants/identity-provider-constants.ts]: Improved with 520dba0695d501fe0ca61d7edb365d58ea7cf899.

**Review by brionmario — APPROVED** [review]

**Pull request merged**